test/clojure-mode-external-interaction-test.el

```diff
@@ -0,0 +1,108 @@
+;;; clojure-mode-external-interaction-test.el --- Clojure Mode interactions with external packages test suite  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2014-2020 Bozhidar Batsov <bozhidar@batsov.com>
+
+;; This file is not part of GNU Emacs.
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Code:
+
+(require 'clojure-mode)
+(require 'buttercup)
+(require 'paredit)
+
+(describe "Interactions with Paredit"
+  ;; reuse existing when-refactoring-it macro
+  (describe "it should insert a space"
+    (when-refactoring-it "before lists"
```

Comment:
I get the point of using this macro here, but probably we need some alias for it, otherwise the tests read a bit weird. Probably not a big deal, though, as I can't suggest a great name right now - something like `with-edits/actions`.

Reply:
I couldn't think of a name as well, but at least the console output makes sense -

```
Interactions with Paredit:
  it should insert a space
    before lists (0.19ms)
    before vectors (0.17ms)
    before maps (0.16ms)
    before strings (0.15ms)
    after gensym (0.20ms)
    after symbols ending with ' (0.19ms)
  it should not insert a space
    for anonymous fn syntax (0.19ms)
    for hash sets (0.17ms)
    for regexes (0.16ms)
    for quoted collections (0.17ms)
    for reader conditionals (0.17ms)

```